README.md

```diff
@@ -1 +1,205 @@
+
+`go-ipld-prime` provides a series of builder interfaces for building this kind of structured data into a node. If your library simply wants to make a selector from CID and a path, represented by an array of strings, you could construct the node as follows:
+
+```golang
+import (
+	ipld "github.com/ipld/go-ipld-prime"
+	free "github.com/ipld/go-ipld-prime/impl/free"
+	fluent "github.com/ipld/go-ipld-prime/fluent"
+  cidLink "github.com/ipld/go-ipld-prime/linking/cid"
+)
+
+func SelectorSpecFromCidAndPath(lnk cid.Cid, pathSegments []string) (ipld.Node, error) {
+	var node ipld.Node
+	err := fluent.Recover(func() {
+		builder := fluent.WrapNodeBuilder(free.NodeBuilder())
+		node = builder.CreateMap(func (mb fluent.MapBuilder, knb fluent.NodeBuilder, vnb fluent.NodeBuilder) {
+			mb.Insert(knb.CreateString("root"), vnb.CreateLink(cidLink.Link{lnk}))
+			mb.Insert(knb.CreateString("selectors"), 
+			vnb.CreateList(func (lb fluent.ListBuilder, vnb fluent.NodeBuilder) {
+				for _, pathSegment := range pathSegments {
+					lb.Append(CreateMap(
+						func (mb fluent.MapBuilder, knb fluent.NodeBuilder, vnb fluent.NodeBuilder) {
+							mb.Insert(knb.CreateString("selectPath"), vnb.CreateString(pathSegment))
+						},
+					))
+				}
+			}))
+		});
+	})
+	if err != nil {
+		return nil, err
+	}
+	return node, nil
+}
+```
+
+Alternatively, just call:
+
+```golang
+rootedSelector := graphsync.SelectorSpecFromCidAndPath(lnk, pathSegments)
+```
+
+### Response Type
+
+```golang
+
+type ResponseProgress struct {
+  Node      ipld.Node // a node which matched the graphsync query
+  Path      ipld.Path // the path of that node relative to the traversal start
+	LastBlock struct {  // LastBlock stores the Path and Link of the last block edge we had to load. 
+		ipld.Path
+		ipld.Link
+	}
+}
+
+```
+
+The above provides both immediate and relevant metadata for matching nodes in a traversal, and is very similar to the information provided by a local IPLD selector traversal in `go-ipld-prime`
+
+## Compatibility: Block Requests
+
+While the above is extremely useful if your library already uses `go-ipld-prime`, if your library uses an older version of go ipld libraries, working with these types of `go-ipld-prime` data may prove challenging.
+
+To support these clients, Graphsync provides a compatibility version of the above function that returns just blocks that were traversed:
+
+```golang
+var blocksChan <-chan blocks.Block
+var errors <-chan error
+
+blocksChan, errors = exchange.GetBlocks(ctx context.Context, p peer.ID, rootedSelector Node)
+```
+
+This is provided as a transitional layer and `go-graphsync` may drop support for this format in the future.
+
+## Contribute
+
+PRs are welcome!
+
+Small note: If editing the Readme, please conform to the [standard-readme](https://github.com/RichardLitt/standard-readme) specification.
+
+## License
+
+MIT © Protocol Labs
```

Comment:
I believe we are actually a dual MIT & Apache-2 License re https://github.com/ipfs/team-mgmt/issues/849

Reply:
Will do.

BTW, that was a copy from the Bitswap readme so we should probably update that as well.

Reply:
Went ahead and updated and also added both licenses in master (didn't want to pollute this PR with more stuff)